src/main/webapp/scroll-container.js

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Creates a `ScrollContainer` responsible for hiding and displaying
+ * a clickable banner to restart automatic scrolling.
+ *
+ * <p>Automatic scrolling is when the elements in the container are scrolled
+ * to the top of the container automatically.
+ */
+export class ScrollContainer extends HTMLDivElement {
+  static #SCROLL_BANNER_CLASSES =
+      'scroll-banner sticky-top p-2 text-center text-white font-weight-bold';
+  static #SCROLL_CONTAINER_CLASSES = 'mx-5 my-3 bg-light pb-3 rounded';
+  static #AUTO_SCROLL_MESSAGE = 'Click here to continue auto-scroll';
+
+  #autoScrollIsActive;
+  #scrollBanner;
+  browserScrolled;
```

Comment:
why isn't this private?

Reply:
ah I must have missed that, I made it private. Thanks!